test/extensions/quic_listeners/quiche/active_quic_listener_test.cc

```diff
@@ -60,13 +76,21 @@ class ActiveQuicListenerTest : public testing::TestWithParam<Network::Address::I
         connection_handler_(*dispatcher_) {}
 
   void SetUp() override {
+    envoy::config::bootstrap::v3::LayeredRuntime config;
+    config.add_layers()->mutable_admin_layer();
+    loader_ = std::make_unique<Runtime::ScopedLoaderSingleton>(Runtime::LoaderPtr{
+        new Runtime::LoaderImpl(*dispatcher_, tls_, config, local_info_, init_manager_, store_,
+                                generator_, validation_visitor_, *api_)});
     listen_socket_ =
         std::make_shared<Network::UdpListenSocket>(local_address_, nullptr, /*bind*/ true);
     listen_socket_->addOptions(Network::SocketOptionFactory::buildIpPacketInfoOptions());
     listen_socket_->addOptions(Network::SocketOptionFactory::buildRxQueueOverFlowOptions());
 
-    quic_listener_ = std::make_unique<ActiveQuicListener>(
-        *dispatcher_, connection_handler_, listen_socket_, listener_config_, quic_config_, nullptr);
+    quic_listener_ = std::make_unique<ActiveQuicListener>(*dispatcher_, connection_handler_,
+                                                          listen_socket_, listener_config_,
+                                                          quic_config_, nullptr, enabledFlag());
```

Comment:
Overriding enabledFlag() is a bit indirect. Can you add two methods like below:
```   
std::make_unique<ActiveQuicListener> createQuicListenerFactory(const std::string& yaml) {
std::string listener_name = QuicListenerName;
    auto& config_factory =
        Config::Utility::getAndCheckFactoryByName<Server::ActiveUdpListenerConfigFactory>(
            listener_name);
    ProtobufTypes::MessagePtr config = config_factory.createEmptyConfigProto();
    TestUtility::loadFromYaml(yaml, *config);
    return config_factory.createActiveUdpListenerFactory(*config, /*concurrency=*/1);

virtual std::string yamlForQuicConfig() {
  return  R"EOF(
runtime_key: "quic.enabled"
default_value: true
)EOF";
}
```
And use them here:
```
EXPECT_CALL(listener_config_, listenSocketFactory());
EXPECT_CALL(listener_config_. socket_factory_, getListenSocket).WillOnce(Return(listen_socket_));
quic_listener_ = createQuicListenerFactory(yamlForQuicConfig())->createActiveUdpListener(*dispatcher_, connection_handler_, ...);
```

In ActiveQuicListenerEmptyFlagConfigTest, 
```
std::string yamlForQuicConfig() override {
 return R"EOF(
    max_concurrent_streams: 10
  )EOF";
}
```